### DISCUSSION TITLE
Database migration error when setting up self-hosted

### DISCUSSION THREAD

**Discussion by jceddy** [keygen-sh/community#145] [Issue]
I'm setting up a self-hosted keygen instance, using an existing pgsql server (creating a new user/database for keygen).

When the database migrations run, I get the following errors:

```
rails aborted!
StandardError: Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Active Record release the migration was written for:

class RolifyCreateRoles < ActiveRecord::Migration[7.0]
/app/db/migrate/20160927003417_rolify_create_roles.rb:3:in `<top (required)>'

Caused by:
StandardError: Directly inheriting from ActiveRecord::Migration is not supported. Please specify the Active Record release the migration was written for:

class RolifyCreateRoles < ActiveRecord::Migration[7.0]
/app/db/migrate/20160927003417_rolify_create_roles.rb:3:in `<top (required)>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```

I didn't have this issue in my previous test setups using containerized pgsql server. I'm wondering whether anyone can shed some light on what might be different in the non-containerized pgsql instance causing this error? (and how to fix it)

**Comment by ezekg** [comment]
You seem to be migrating the database from zero, i.e. a new database. This isn't supported. Instead, you should run `rails db:schema:load` to load the schema without running migrations. Migrations are only meant to be run on an existing database, not on a new database.

Just FYI, this should be run for you when running [the recommended `rake keygen:setup` command](https://github.com/keygen-sh/keygen-api/blob/master/lib/tasks/keygen/setup.rake#L5).
